clippy_lints/src/large_enum_variant.rs

```diff
@@ -67,32 +71,50 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for LargeEnumVariant {
                             }
                         })
                         .sum();
-                    if size > self.maximum_variant_size_allowed {
-                        span_lint_and_then(cx,
-                                           LARGE_ENUM_VARIANT,
-                                           def.variants[i].span,
-                                           "large enum variant found",
-                                           |db| {
-                            if variant.fields.len() == 1 {
-                                let span = match def.variants[i].node.data {
-                                    VariantData::Struct(ref fields, _) |
-                                    VariantData::Tuple(ref fields, _) => fields[0].ty.span,
-                                    VariantData::Unit(_) => unreachable!(),
-                                };
-                                if let Some(snip) = snippet_opt(cx, span) {
-                                    db.span_suggestion(span,
-                                                       "consider boxing the large fields to reduce the total size of \
-                                                        the enum",
-                                                       format!("Box<{}>", snip));
-                                    return;
-                                }
-                            }
-                            db.span_help(def.variants[i].span,
-                                         "consider boxing the large fields to reduce the total size of the enum");
-                        });
-                    }
+
+                    sizes.push(size);
+                    variants.push(variant);
                 });
             }
+
+            let mut grouped = sizes.into_iter().zip(variants.into_iter().enumerate()).collect::<Vec<_>>();
```

Comment:
instead of collecting all vectors, couldn't you just have the previous for loop extract the minimum and maximum value?